src/main/java/fi/aalto/cs/apluscourses/intellij/activities/InitializationActivity.java

```diff
@@ -70,6 +64,8 @@ public void runActivity(@NotNull Project project) {
     }
     PluginSettings.getInstance()
         .getMainViewModel(project).courseViewModel.set(new CourseViewModel(course));
+    MainViewModelUpdater mainViewModelUpdater = new MainViewModelUpdater(project);
+    new Thread(mainViewModelUpdater::run).start();
```

Comment:
Are there two threads of this kind? 😕 Is it possible to make it so that there is only one place in the code where this updater is spawned so that is easier to understand when does that happen?

Reply:
There should be only one thread, but there are two different cases:
  + The project isn't a course project, so no thread created by the initialization activity. A thread may be created later by the course project action.
  + The project is a course project, so a thread is created in the initialization activity.

I agree it would be simpler if there's only one place. I'll see if I can refactor it.